packager/react-native-xcode.sh

```diff
@@ -46,7 +46,14 @@ fi
 # npm global install path may be a non-standard location
 PATH="$(npm prefix -g)/bin:$PATH"
 
-react-native bundle \
+# check for NPM-installed react-native-cli
+if [[ -d ./node_modules/react-native-cli ]] && [ -z "$REACT_NATIVE_PATH" ]; then
+  REACT_NATIVE_PATH=./node_modules/.bin/react-native
```

Comment:
FYI: this is a dummy binary that warns users that they have installed `react-native` globally (instead of `react-native-cli`)